aesara/tensor/basic_opt.py

```diff
@@ -155,7 +155,7 @@ def broadcast_like(value, template, fgraph, dtype=None):
     rval = alloc(value, *new_shape)
     # the template may have 1s in its shape without being broadcastable
     if rval.broadcastable != template.broadcastable:
-        rval = unbroadcast(
+        rval = mask_broadcastable(
```

Comment:
This is the only other use of "unbroadcasting" outside of Scan. One more reason to address https://github.com/aesara-devs/aesara/issues/159

Reply:
Trying to revive #288

Reply:
The trouble with replacing `broadcast_like` with `BroadcastTo`s is that at least a few uses of `broadcast_like` require that the resulting broadcasted array be writable.  We currently don't have a means of converting the non-writable `BroadcastTo`s to writable ones (e.g. a rewrite that adds a `DeepCopy` or something), so introducing too many `BroadcastTo`s can limit the amount of in-placing that's possible.